be/src/storage/rowset/beta_rowset_writer.cpp

```diff
@@ -57,25 +57,6 @@ BetaRowsetWriter::BetaRowsetWriter(const RowsetWriterContext& context)
           _total_index_size(0) {}
 
 Status BetaRowsetWriter::init() {
-    DCHECK(!(_context.tablet_schema->contains_format_v1_column() &&
-             _context.tablet_schema->contains_format_v2_column()));
-    auto real_data_format = _context.storage_format_version;
-    auto tablet_format = real_data_format;
-    if (_context.tablet_schema->contains_format_v1_column()) {
-        tablet_format = kDataFormatV1;
-    } else if (_context.tablet_schema->contains_format_v2_column()) {
-        tablet_format = kDataFormatV2;
-    }
-
-    // StarRocks is built on earlier work on Apache Doris and is compatible with its data format but
-    // has newly designed storage formats for DATA/DATETIME/DECIMAL for better performance.
-    // When loading data into a tablet created by Apache Doris, the real data format will
-    // be different from the tablet schema, so here we create a new schema matched with the
-    // real data format to init `SegmentWriter`.
-    if (real_data_format != tablet_format) {
```

Comment:
This code can be removed, because we have some users' data come from Doris, their tablet meta is still format v1. 
You can only delete this after changed their tablet meta

Reply:
oh, yeah, changed in e3888979498b6838acf2f1e23b742da587724d5f

Reply:
oh, yeah, changed in 3512784e575f4af624d4a5a2b2455d9eeb0e32c6